GoogleChromeLabs/benchmark-workloads
# Add workloads-testing-utils

This pr enables a workload to run its own interaction tests.

A new `workload-testing-utils` package has been added, that a workload can use in combination with a `workload.test.js` file. 

Example for `news-site-next`:

```
window.suites = new BenchmarkTestSuites(window.name, [
    new BenchmarkTestSuite("Navigation", [
        new BenchmarkTestStep("Navigate to US page", async () => {
            await runWorkloadTest(() => document.querySelector("#navbar-navlist-us-link").click());
        }),
        new BenchmarkTestStep("Navigate to World page", async () => {
            await runWorkloadTest(() => document.querySelector("#navbar-navlist-world-link").click());
        }),
        new BenchmarkTestStep("Navigate to Politics page", async () => {
            await runWorkloadTest(() => document.querySelector("#navbar-navlist-politics-link").click());
        }),
    ]),
    new BenchmarkTestSuite("Dropdown", [
        new BenchmarkTestStep("Toggle More Dropdown", async () => {
            await runWorkloadTest(() => document.querySelector("#navbar-dropdown-toggle").click());
        }),
        new BenchmarkTestStep("Toggle More Dropdown", async () => {
            await runWorkloadTest(() => document.querySelector("#navbar-dropdown-toggle").click());
        }),
    ]),
]);
```

how to test locally:
```
pnpm install
pnpm -F news-site-next start:static
```

Open your browser and type any of the following in your console:
```
await window.suites.run(["Navigation", "Dropdown"]);
await window.suites.run(["Dropdown"]);
await window.suites.run();
```

Note:
This does not yet do its own performance measurement yet.
Its expected to be used with static workloads at the moment.

Next step:
change speedometer branch to use this setup.

@kara
